README.md

```diff
@@ -1,2 +1,285 @@
+[Apache Airflow](http://airflow.apache.org/) and [Ansible](https://docs.ansible.com/).
+
+Queenbee populates and validates the workflows but does not run them. For running the
+workflows see `ladybug-tools/workerbee` which converts Queenbee workflows to executable
+[Luigi](https://luigi.readthedocs.io/en/stable/) workflows.
+
+You can find more workflow samples in
+[honeybee-radiance-workflow](https://github.com/ladybug-tools/honeybee-radiance-workflow)
+repository. The workflow is meant to mainly be used with Ladybug Tools libraries but it
+should also support any other custom workflows.
+
+# Queenbee Workflow structure
+
+A Queenbee workflow is a YAML / JSON file which is consisted from different parts. You
+can use Queenbee to generate the workflows programmatically using Python or you can
+write a workflow line by line.
+
+Here we discuss the generic structure of a Queenbee workflow. See OpenApi schema
+documentation for all Workflow objects: [Missing link]()
+
+```
+
+  Workflow
+    |
+    |__ inputs
+    |   |___ parameters
+    |   |___ artifacts
+    |
+    |__ operators
+    |
+    |__ templates
+    |
+    |__ flow
+    |   |___ tasks / steps
+    |
+    |__ outputs
+        |___ parameters
+        |___ artifacts
+
+```
+
+## 1. inputs
+
+Workflow inputs are global inputs that can be accessed using `"{{workflow.inputs.xx.yy}}"`.
+For instance to access an input parameter with the name `file_path` one should use
+`"{{workflow.inputs.parameters.file_path}}"` and Queenbee will map the value of
+`file_path` parameter to this placeholder.
+
+Inputs can be from two different types: `parameters` and `artifacts`. Use `artifacts` for
+*files* and *folder* inputs. The content of these `artifacts` will be copied to target
+location before executing the workflow. For all the other `inputs` use `parameters`.
+
+Here is an example `inputs` for a daylight simulation workflow. You can put a default
+value for inputs or it can be passed to workflow using an input YAML / JSON file.
+
+```yaml
+
+  inputs:
+    parameters:
+    # this is just the path to target folder not the content itself. It can be referenced
+    # as {{workflow.parameters.folder}}
+    - name: folder
+      value: /mnt/c/ladybug/untitled
+    artifacts:
+    - name: sky_file
+      value: full_path_to_sky_file.sky
+    - name: material_file
+      value: full_path_to_material_file.mat
+    - name: geometry_file
+      value: full_path_to_geometry_file.rad
+
+```
+
+## 2. operators
+
+`Operators` include the requirements for running `templates` locally or using a Docker
+image. This is an operator for running arbitrary Radiance commands.
+
+```yaml
+
+operators:
+  - name: radiance-operator
+    inputs:
+      parameters:
+        - name: command
+    container:
+      name: radiance52
+      image: ladybugtools/radiance:5.2
+      command:
+        - /bin/sh
+        - -c
+      args:
+        - '{{ inputs.parameters.command }}'
+    requirements:
```

Comment:
Does this requirements section check for specific versions in the current system or does it also try to install it for you?

Reply:
This is out of the scope of `queenbee` and should be decided when one runs the study. I have been thinking it should be an option in the command.

`workerbee [WORKFLOW] [INPUT_FILE] --install-requirements`

Reply:
My main reason for bringing this up is because it seems very ambitious to include automated install of arbitrary software.

Reply:
Good point probably we should add a section for user to provide installation commands or otherwise throw an exception. Added an issue to address this in the future.